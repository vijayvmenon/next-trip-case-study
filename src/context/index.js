import { useContext } from 'react';
import AppContext, { Context } from './context';

export default AppContext;
export { Context };
export const useMyContext = () => useContext(Context);
