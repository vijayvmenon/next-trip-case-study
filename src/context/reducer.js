const initialState = {
  tableExpanded: false,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'TABLE_EXPAND':
      return { ...state, tableExpanded: true };
    case 'TABLE_COLLAPSE':
      return { ...state, tableExpanded: false };
    default:
      return { ...state };
  }
}

export { initialState };
