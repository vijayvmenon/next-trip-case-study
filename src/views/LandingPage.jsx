import React, { useState, useEffect } from 'react';
import { createUseStyles } from 'react-jss';
import constants, { colors } from '../constants';
import SearchOptions from './SearchOptions';
import RouteSearch from './RouteDirectionStopSearch';
import StopSearch from './StopNumberSearch';

const useStyles = createUseStyles({
  root: {
    flexGrow: 1,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  appHeaderText: {
    color: colors.GREY,
  },
});

export default function LandingPage() {
  const classes = useStyles();
  const [options, setOptions] = useState([
    {
      name: constants.BY_ROUTE,
      hrefLink: '/byRoute',
      active:
        window.location.pathname === '/' ||
        window.location.pathname === '/byRoute',
      component: <RouteSearch />,
    },
    {
      name: constants.BY_STOP,
      hrefLink: '/byStop',
      active: window.location.pathname === '/byStop',
      component: <StopSearch />,
    },
  ]);

  useEffect(() => {
    if (window.location.pathname === '/') {
      window.location.href = '/byRoute';
    }
  }, []);

  const setActiveOption = (activeVal) => {
    setOptions(
      options.map((val) => ({
        ...val,
        active: val.name === activeVal,
      })),
    );
  };

  return (
    <div className={classes.root}>
      <header>
        <h2 className={classes.appHeaderText}>{constants.APP_HEADER}</h2>
      </header>
      <SearchOptions options={options} click={setActiveOption} />
      {options.find((val) => val.active).component}
    </div>
  );
}
