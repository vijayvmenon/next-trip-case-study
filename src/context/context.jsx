import React, { useReducer, createContext } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';

export const Context = createContext();

export default function AppContext({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
}

AppContext.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,
};
