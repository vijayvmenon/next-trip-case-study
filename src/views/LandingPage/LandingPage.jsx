import React, { useEffect, useContext } from 'react';
import { createUseStyles } from 'react-jss';
import { Switch, Route, useLocation } from 'wouter';
import constants, { colors } from '../../constants';
import SearchOptions from '../SearchOptions/SearchOptions';
import RouteSearch from '../ByRouteSearch/RouteDirectionStopSearch';
import StopSearch from '../ByStopSearch/StopNumberSearch';
import PageNotFound from '../../components/PageNotFound/PageNotFound';
import { Context } from '../../context';

const useStyles = createUseStyles({
  root: {
    flexGrow: 1,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  appHeaderText: {
    color: colors.GREY,
  },
});

export default function LandingPage() {
  const classes = useStyles();
  const [location, setLocation] = useLocation();
  const { state: { hideHeader, translations = {} } = {} } = useContext(Context);
  useEffect(() => {
    if (
      location === '/' ||
      location === constants.BASE_PATH ||
      location === constants.BASE_PATH_2
    ) {
      setLocation(`${constants.BASE_PATH}/byRoute`);
    }
  }, [location, setLocation]);

  return (
    <div className={classes.root}>
      {!hideHeader && (
        <>
          <header>
            <h2 className={classes.appHeaderText}>{translations.APP_HEADER}</h2>
          </header>
          <SearchOptions options={constants.NAVIGATION_OPTIONS} />{' '}
        </>
      )}
      <Switch>
        <Route
          path={`${constants.BASE_PATH}/byRoute`}
          component={RouteSearch}
        />
        <Route path={`${constants.BASE_PATH}/byStop`} component={StopSearch} />
        <Route
          path={`${constants.BASE_PATH}/byRoute/:routeId/:directionId/:stopId`}
          component={RouteSearch}
        />
        <Route
          path={`${constants.BASE_PATH}/byStop/:stopId`}
          component={StopSearch}
        />
        <Route>
          <PageNotFound />
        </Route>
      </Switch>
    </div>
  );
}
