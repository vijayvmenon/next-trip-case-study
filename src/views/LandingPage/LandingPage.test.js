import { render, fireEvent, screen } from '@testing-library/react';
import LandingPage from './LandingPage';
import renderer from 'react-test-renderer';
import { Context } from '../../context';

const dispatch = jest.fn(); // mock dispatch

describe('Component:LandingPage.jsx', () => {
  test('LandingPage.jsx renders correctly', () => {
    const landingPage = renderer
      .create(
        <Context.Provider value={{ dispatch }}>
          <LandingPage />
        </Context.Provider>,
      )
      .toJSON();
    expect(landingPage).toMatchSnapshot();
  });
});
