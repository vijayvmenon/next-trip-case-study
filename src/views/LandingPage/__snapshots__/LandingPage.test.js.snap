// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:LandingPage.jsx LandingPage.jsx renders correctly 1`] = `
<div
  className="root-0-2-1"
>
  <header>
    <h2
      className="appHeaderText-0-2-2"
    />
  </header>
  <nav
    className="root-0-2-3"
  >
    <a
      className="linkButtonRoot-0-2-4 undefined"
      data-testid="custom-anchor-"
      href="/next-trip-case-study/byRoute"
      onClick={[Function]}
      to={null}
    >
      
    </a>
    <a
      className="linkButtonRoot-0-2-4 undefined"
      data-testid="custom-anchor-"
      href="/next-trip-case-study/byStop"
      onClick={[Function]}
      to={null}
    >
      
    </a>
  </nav>
   
  <div
    className="root-0-2-6"
  >
    <div
      className="firstLine-0-2-7"
    >
       
      <svg
        className="icon-0-2-8"
      >
        app-logo.svg
      </svg>
      <span
        className="notFound404-0-2-9"
      >
        404
      </span>
    </div>
    <div
      className="secondLine-0-2-11"
    />
    <div
      className="thirdLine-0-2-12"
    />
    <div
      className="buttonDiv-0-2-10"
      data-testid="go-home-button"
    >
      <button
        className="root-0-2-13"
        data-testid="custom-button"
        onClick={[Function]}
        type="button"
      >
        
      </button>
    </div>
  </div>
</div>
`;
